src/routes/wishlistRoutes.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const Wishlist = require('../models/Wishlist');
+const Album = require('../models/Album');
+const router = express.Router();
+const mongoose = require('mongoose');
+
+router.patch('/wishlist/:wishlist_id/add_album/:album_id', async (req, res) => {
+  const { wishlist_id, album_id } = req.params;
+  if (
+    !mongoose.Types.ObjectId.isValid(wishlist_id) ||
+    !mongoose.Types.ObjectId.isValid(album_id)
+  ) {
+    return res.status(400).send('Invalid wishlist_id or album_id');
+  }
+  // Find the Wishlist and Album documents
+  const wishlist = await Wishlist.findById(wishlist_id);
+  const album = await Album.findById(album_id);
+  // Check if wishlist and album exist
+  if (!wishlist || !album) {
+    return res.status(404).send('Wishlist or album not found');
+  }
+  // Check if album is already in wishlist
+  if (wishlist.albums.includes(album._id)) {
+    return res.status(200).json(wishlist);
+  }
+  // Add album to wishlist and save
+  wishlist.albums.push(album._id);
+  await wishlist.save();
+  return res.status(200).json(wishlist);
+});
```

Comment:
We defined this in the controller, so we should call out to the controller here:

```suggestion
router.patch('/wishlist/:wishlist_id/add_album/:album_id', addAlbumToWishlist);
```

Reply:
Maybe also use the `authenticateUser` middleware here and for the remove endpoint to ensure a logged in user is calling this endpoint?